src/btree/bt_cursor.c

```diff
@@ -888,6 +908,49 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
+/*
+ * __btcur_search_near_row_pinned_page --
+ *     Search a row store pinned page before searching from the root in search near.
+ */
+static inline int
+__btcur_search_near_row_pinned_page(WT_CURSOR_BTREE *cbt, bool *validp)
```

Comment:
If I understand correctly pulling this into it's own function is just refactoring/simplifying the code. There isn't a functional change here.

Reply:
Yeah this was just trying to make search_near a bit less confusing for the average reader.